tracer/src/Datadog.Trace/Iast/Location.cs

```diff
@@ -9,6 +9,22 @@ namespace Datadog.Trace.Iast;
 
 internal readonly struct Location
 {
+    public Location(string method)
+    {
+        int index = method.LastIndexOf("::");
+        if (index >= 0)
+        {
+            Path = method.Substring(0, index);
+            Method = method.Substring(index + 2);
+            index = Method.IndexOf("(");
+            if (index > 0) { Method = Method.Substring(0, index); }
+        }
+        else
+        {
+            Method = method;
+        }
```

Comment:
`IndexOf` etc do culture specific comparisons by default, so we should use `Ordinal` here. Also, you can remove one of the Substring allocations I think 🙂 
```suggestion
        var index = method.LastIndexOf("::", StringComparison.Ordinal);
        if (index >= 0)
        {
            Path = method.Substring(0, length: index);
            var bracketIndex = method.IndexOf("(", startIndex: index + 2, StringComparison.Ordinal);
            Method = bracketIndex > 0
                         ? method.Substring(index + 2, length: bracketIndex - index - 2)
                         : method.Substring(index + 2);
        }
        else
        {
            Method = method;
        }
```